sdks/java/extensions/avro/src/main/java/org/apache/beam/sdk/extensions/avro/schemas/utils/AvroUtils.java

```diff
@@ -1050,17 +1050,22 @@ private static org.apache.avro.Schema getFieldSchema(
         String type = options.hasOption("type") ? options.getValue("type") : "bytes";
         Type decimalType = Type.valueOf(type.toUpperCase());
         Integer precision =
-                options.hasOption("precision") ? options.getValue("precision") : Integer.MAX_VALUE;
+            options.hasOption("precision") ? options.getValue("precision") : Integer.MAX_VALUE;
         Integer scale = options.hasOption("scale") ? options.getValue("scale") : 0;
         LogicalTypes.Decimal decimal = LogicalTypes.decimal(precision, scale);
         if (decimalType == Type.FIXED) {
+          if (!options.hasOption("size")) {
+            throw new IllegalArgumentException("'size' option is required for field:" + fieldName);
```

Comment:
Not familiar with FIXED types, but does it make sense to set a default value instead of throwing an exception?